.github/workflows/regression-tests.yml

```diff
@@ -0,0 +1,472 @@
+        env:
+          IGNORE: ${{ fromJSON(steps.config.outputs.repository).ignore }}
+        run: |
+          echo "$IGNORE" >> .gitignore
+          git rm --cached -r $(git ls-files -i -c --exclude-from=.gitignore)
+          git clean -fdX
+      - name: Configure Hardhat
+        if: startsWith(matrix.command, 'hardhat')
+        env:
+          CONFIG: ${{ fromJSON(steps.config.outputs.repository).hardhat-config }}
+        run: |
+          rm hardhat.config.* || true
+          echo "$CONFIG" > hardhat.config.js
+      - name: Initialize the package
+        if: startsWith(matrix.command, 'hardhat')
+        run: |
+          npm init -y
+          npm pkg set type="module"
+      - name: Install dependencies (npm)
+        if: hashFiles('package-lock.json') != ''
+        run: npm ci
+      - name: Install dependencies (pnpm)
+        if: hashFiles('pnpm-lock.yaml') != ''
+        run: |
+          npm install -g pnpm
+          pnpm install --frozen-lockfile --prefer-offline || pnpm install
+      - name: Install dependencies (yarn)
+        if: hashFiles('yarn.lock') != ''
+        run: |
+          npm install -g yarn
+          yarn install --immutable
+      - name: Install dependencies (bun)
+        if: hashFiles('bun.lockb') != ''
+        run: |
+          npm install -g bun
+          bun install --frozen-lockfile
+      - name: Run ${{ matrix.command }}
+        id: run
+        env:
+          DEBUG: ${{ runner.debug && '*' || '' }}
+          ENV_SOURCE: ${{ fromJSON(steps.config.outputs.repository).env }}
+        run: |
+          source <(echo "$ENV_SOURCE")
+          set +e
+          before=$(date +%s)
+          ${{ matrix.command }} 2>&1 | tee -a run.output
+          status=${PIPESTATUS[0]}
+          after=$(date +%s)
+          set -e
+          echo '{"status": '"$status"', "time": '"$((after - before))"'}' > run.result
+      - name: Run the command post-hook
+        uses: actions/github-script@v7
+        env:
+          PATTERN: ${{ fromJSON(steps.config.outputs.command).pattern }}
+          TEMPLATE: ${{ fromJSON(steps.config.outputs.command).template }}
+        with:
+          script: |
+            const fs = require('fs');
+            const path = require('path');
+            const output = fs.readFileSync('run.output').toString();
+            const matches = output.matchAll(new RegExp(process.env.PATTERN, 'g'));
+            const sums = [];
+            for (const groups of matches) {
+              for (let i = 1; i < groups.length; i++) {
+                while (sums.length < i) {
+                  sums.push(0);
+                }
+                sums[i - 1] += parseInt(groups[i]);
+              }
+            }
+            let details = process.env.TEMPLATE;
+            for (let i = 0; i < sums.length; i++) {
+              details = details.replaceAll(`\${${i}}`, sums[i]);
+            }
+            fs.writeFileSync('run.details', details);
```

Comment:
This scripts extracts details about the command run from its output using the command specific pattern that was provided in the config. It sums the numbers extracted from the output using the pattern and feeds them to the command specific template to get a nice and short details summary - something like `Compiled 10 files`.